### PR TITLE
Add resource server config to secure api endpoints

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

Adds resource server configuration for creating secure endpoints. 

Adds a sample endpoint for testing in dev environment. response looks like 

```
{
    "client": "interventions",
    "roles": [
        "ROLE_INTERVENTIONS",
        "ROLE_AUTH_DELIUS_LDAP",
        "ROLE_COMMUNITY"
    ],
    "tokenExpiry": "2020-11-26T12:11:00Z"
}
```

## What is the intent behind these changes?

scaffolding to allow development of secure endpoints

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,14 +1,27 @@
 version: "3"
 services:
-  hmpps-interventions-service:
-    build:
-      context: .
-    network_mode: "host"
-    container_name: hmpps-interventions-service
+  hmpps-auth:
+    image: quay.io/hmpps/hmpps-auth:latest
```
Comment (sldblog) [inline docker-compose.yml:4]: Very generic question, sorry. Is it safe to assume `latest` is the most recent build on the `main` branch?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityApiService.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class CommunityApiService(private val communityApiWebClient: WebClient) {
+  fun <T> emptyWhenNotFound(exception: WebClientResponseException): Mono<T> = emptyWhen(exception, HttpStatus.NOT_FOUND)
+  fun <T> emptyWhen(exception: WebClientResponseException, statusCode: HttpStatus): Mono<T> =
+    if (exception.rawStatusCode == statusCode.value()) Mono.empty() else Mono.error(exception)
+
+  fun getDetailsByUsername(username: String): User? {
+    return communityApiWebClient.get()
+      .uri("/secure/users/$username/details")
+      .retrieve()
+      .bodyToMono(User::class.java)
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityApiService.kt:20]: We'll most likely have CRN lookups from the probation offender search (possibly in the UI?), so we may not use this endpoint. Is it ok to remove the community API service until we have its first use case from a story?

**Review by sldblog — COMMENTED** [review]
I'm wondering if we should wait for the community-api bits on the first user story that requires it? The (read) API _may_ not be used in this app, which could lead to unused code.

**Comment by tomsmyers** [comment]
@sldblog i added the community api endpoint purely to demo that auth is working e2e. no intention of keeping around if not needed.

File: docker-compose.yml (hunk 1 above)
Comment (tomsmyers) [reply docker-compose.yml:4]: that's right

**Review by sldblog — APPROVED** [review]

**Comment by sldblog** [comment]
Sorry for taking so long!

**Pull request merged**